Hardware/Hardware/CardReader/CardReaderServiceError.swift

```diff
@@ -1,20 +1,152 @@
 /// Models errors thrown by the CardReaderService.
-/// This is doing the bare minimum for now.
-/// Proper error handling is coming in
-/// https://github.com/woocommerce/woocommerce-ios/issues/3734
+/// It identifies the interaction with the card reader
+/// where the error was thrown.
+/// It is important to mark the specific interaction
+/// because some operations (like, for example, procesing a payment)
+/// require of three interactions with the SDK in sequence, and any error
+/// in any of the steps in that sequence would make the whole operation fail
 public enum CardReaderServiceError: Error {
     /// Error thrown during reader discovery
-    case discovery
+    case discovery(underlyingError: UnderlyingError = .internalServiceError)
 
     /// Error thrown while connecting to a reader
-    case connection
+    case connection(underlyingError: UnderlyingError = .internalServiceError)
 
     /// Error thrown while creating a payment intent
-    case intentCreation
+    case intentCreation(underlyingError: UnderlyingError = .internalServiceError)
 
     /// Error thrown while collecting payment methods
-    case paymentMethod
+    case paymentMethod(underlyingError: UnderlyingError = .internalServiceError)
 
     /// Error thrown while capturing a payment
-    case capturePayment
+    case capturePayment(underlyingError: UnderlyingError = .internalServiceError)
```

Comment:
Similarly, could we make this more like the naming order of the others and make this 'paymentCapture` ?